### PR TITLE
Avoid player overlapping crystals and enemies.

### PR DESCRIPTION
**Pull request by bernatgb**

### CODE REVIEW

File: Source/src/components/ComponentAgent.cpp [hunk 1]
```diff
@@ -32,15 +36,36 @@ void Hachiko::ComponentAgent::Update()
...
 {
     AddToCrowd();
+    if (player)
+    {
+        dtCrowdAgent* agent = App->navigation->GetEditableAgent(agent_id);
+        if (!agent)
+        {
+            return;
+        }
+        agent->params.radius = 2.0f;
```
Comment (miquelmiro3) [inline Source/src/components/ComponentAgent.cpp:67]: We should try to hardcode number (magic numbers).

**Comment by miquelmiro3** [comment]
Hey tot bé. Només un parell de coses. Com el parasit de drop forma part del enemic el player el pot empuxar ja que es un agent igual que el drop. Suposo que es més un problema del que el drop forma part del enemic actualment quan s'hauria de instanciar però s'hauria de mirar potser abans de fer merge. També una pregunta per curiositat, no es podia fer que el player fos un obstacle més que un agent?

**Comment by miquelmiro3** [comment]
> 

I am used to talk to u in catalan, my bad. Traduction:

Just a couple of things. As the drop parasite is part of the enemy the player can push it as it is an agent just like the drop. I guess it's more of a problem of the drop being part of the enemy when it should be instantiated, but we should try to fix that  before merging. Also a question out of curiosity, couldn't the player be made more of an obstacle than an agent?

File: Source/src/components/ComponentAgent.h [hunk 2]
```diff
@@ -48,6 +49,8 @@ namespace Hachiko
 		bool use_pathfinder = true;
 
 		bool show_debug_info = false;
+
+		bool player = false;
```
Comment (baransrc) [inline Source/src/components/ComponentAgent.h:53]: Let's rename this to is_player, the name "player" feels like a non-boolean object.

File: Source/src/components/ComponentAgent.cpp [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
 Hachiko::ComponentAgent::ComponentAgent(GameObject* container) : Component(Type::AGENT, container)
 {
+    if (game_object->name == "PlayerN")
+    {
+        player = true;
+    }
```
Comment (baransrc) [inline Source/src/components/ComponentAgent.cpp:18]: We shouldn't have this hard-coded by the name of the game object, engine should be as much as abstracted from the game it runs. Let's make a method of component agent that is: ComponentAgent::SetAsPlayer(bool is_player),  and we set the flag accordingly. If anything needs to be rebuilt based on is_player flag, we can set the appropriate variables based on the flag passed to SetAsPlayer.

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the work!! We need some changes, after those are adressed, we are good to go!! 💯

**Review by baransrc — APPROVED** [review]
Everything works fine, I've added some changes and a little fix. Thanks for the great work 💯

**Review by Erick9Thor — APPROVED** [review]

**Pull request merged**